File: bbarena-model/src/main/java/net/sf/bbarena/model/event/game/PlayerPitchStatusEvent.java
package net.sf.bbarena.model.event.game;

import net.sf.bbarena.model.Arena;
import net.sf.bbarena.model.team.Player;
import net.sf.bbarena.model.team.PlayerPitchStatus;
import net.sf.bbarena.model.util.Concat;
import net.sf.bbarena.model.util.Pair;

public class PlayerPitchStatusEvent extends PlayerEvent {

	private static final long serialVersionUID = 1790786618254181381L;

	private PlayerPitchStatus _pitchStatus = null;
	private PlayerPitchStatus _oldPitchStatus = null;

    public PlayerPitchStatusEvent(long playerId) {
        super(playerId);
    }

    public PlayerPitchStatusEvent(long playerId, PlayerPitchStatus pitchStatus) {
        super(playerId);
        _pitchStatus = pitchStatus;
    }

    public PlayerPitchStatus getPitchStatus() {
        return _pitchStatus;
    }

    public void setPitchStatus(PlayerPitchStatus pitchStatus) {
        _pitchStatus = pitchStatus;
    }

	@Override
	public void doEvent(Arena arena) {
		Player player = getPlayer(arena);
		_oldPitchStatus = player.getPitchStatus();
		player.setPitchStatus(_pitchStatus);
	}

	@Override
	public void undoEvent(Arena arena) {
		Player player = getPlayer(arena);
		player.setPitchStatus(_oldPitchStatus);
	}

	@Override
	public String getString() {
		return Concat.buildLog(getClass(),
				new Pair("playerId", getPlayerId()),
				new Pair("pitchStatus", _pitchStatus));
	}

}

File: src/main/kotlin/bbarena/model/event/game/PlayerPitchStatusEvent.kt
package bbarena.model.event.game

import bbarena.model.Arena
import bbarena.model.team.PlayerPitchStatus
import bbarena.model.util.Concat

class PlayerPitchStatusEvent : PlayerEvent {

    var pitchStatus: PlayerPitchStatus? = null
    private var _oldPitchStatus: PlayerPitchStatus? = null

    constructor(playerId: Long) : super(playerId) {}

    constructor(playerId: Long, pitchStatus: PlayerPitchStatus) : super(playerId) {
        this.pitchStatus = pitchStatus
    }

    public override fun doEvent(arena: Arena) {
        val player = getPlayer(arena)
        _oldPitchStatus = player.pitchStatus
        player.pitchStatus = pitchStatus
    }

    public override fun undoEvent(arena: Arena) {
        val player = getPlayer(arena)
        player.pitchStatus = _oldPitchStatus
    }

    override fun getString(): String {
        return Concat.buildLog(javaClass,
                Pair("playerId", getPlayerId()),
                Pair("pitchStatus", pitchStatus))
    }

    companion object {

        private val serialVersionUID = 1790786618254181381L
    }

}
